lib/license_finder/license/text.rb

```diff
@@ -6,15 +6,37 @@ module Text
       SPACES = /\s+/.freeze
       QUOTES = /['`"]{1,2}/.freeze
       PLACEHOLDERS = /<[^<>]+>/.freeze
+      SPECIAL_SINGLE_QUOTES = /[‘’]/.freeze
+      SPECIAL_DOUBLE_QUOTES = /[“”„«»]/.freeze
+      ALPHABET_ORDERED_LIST = /\\\([a-z]\\\)\\\s/.freeze
+      ALPHABET_ORDERED_LIST_OPTIONAL = '(\([a-z]\)\s)?'
+      LIST_BULLETS = /(\d{1,2}\\\.|\\\*)\\\s/.freeze
+      LIST_BULLETS_OPTIONAL = '(\d{1,2}.|\*)?\s*'
+      NEWLINE_CHARACTER = /\n+/.freeze
+      QUOTE_COMMENT_CHARACTER = /^\s*\>+/.freeze
+      ESCAPED_QUOTES = /\\\"/.freeze
 
       def self.normalize_punctuation(text)
-        text.gsub(SPACES, ' ')
+        text.dup.force_encoding('UTF-8')
```

Comment:
force encoding does not change the string data but just makes sure that e.g. raw data is read as utf-8. But in our cases we have some text with invalid bytes for utf-8 so this will fail. I need to use

```suggestion
        text.dup.encode('UTF-8', invalid: :replace, undef: :replace, replace: '?')
```

would this also be a possible general solution?

Reply:
Unfortunately, `.encode` is insufficient in some cases. Specifically, I'm trying to add handling for smart quotes. I've been running my tests against this specific license: https://github.com/gregjones/httpcache/blob/master/LICENSE.txt

I did a quick test where I dropped some non-UTF-8 characters into that license and the `force_encoding` seems to handle it nicely. Is there a specific license that you're looking at where it fails?

Reply:
Don't worry, I've found a way that it works with this one. I have an additional `RailsVendor` package manager (to detect `RailsVendorPackage`) because I am to lazy to add those dependencies as "hidden" ones. For that I parse the vendor file itself for package information like version and license. Just have to encode the file as utf-8 itself when reading it.